pyproject.toml

```diff
@@ -1,3 +1,19 @@
+[tool.black]
+exclude = '''
+(
+  /(
+      \.eggs         # exclude a few common directories in the
+    | \.git          # root of the project
+    | \.mypy_cache
+    | \omegaconf/grammar/gen
+    | \.nox
+  )
+)
+'''
+
+[tool.pytest.ini_options]
+addopts = "--import-mode=append"
```

Comment:
Following up on unresolved discussion (https://github.com/omry/omegaconf/pull/321#discussion_r531192207)

@omry I'm not exactly sure what you meant by "Can you try that and see if eliminates the need for this hack?". If you're saying that we could run all tests locally instead (with `pip install -e .`), then yes, we could, but this may hide build issues (ex: files that would be missing with `pip install .` but are present in the local folder).

Reply:
If we `pip install -e .` , we should have the generated grammar files in the current directory and the current directory shold be added to the pythonpath (by pytest).

if we `pip install .`, we should have the generated code installed into the site packages (`/Users/odelalleau/opt/miniconda3/envs/py38-omega/lib/python3.8/site-packages/omegaconf`) and the generated code should be available there.

Can you explain again what problem is this opt solving and in which scenario?

Reply:
So the problem is when using `pip install .`: if we don't set Pytest's import mode to "append", then it inserts the repository's root folder at the *beginning* of `sys.path`, and as a result `import omegaconf` imports the local folder's version instead of the one installed by `pip install .`. By switching to append mode, the repository's root folder is added at the *end* of `sys.path`, and `import omegaconf` correctly imports the installed version.

Reply:
Since the version installed by pip install is identical to the one in the local folder, I guess the problem is that the local folder is missing the generated parser code?
Can we fix this issue by generating the parser code? (either in pip install, or maybe just by running `python setup.py antlr` directly from nox?

Reply:
> Since the version installed by pip install is identical to the one in the local folder, I guess the problem is that the local folder is missing the generated parser code?

Yes

> Can we fix this issue by generating the parser code? (either in pip install, or maybe just by running `python setup.py antlr` directly from nox?

I guess we could, but if we actually want to test the local version it would seem simpler and cleaner to just `pip install -e .` in that case.

Reply:
I meant during it automatically as part of pip install (like in Hydra).

Reply:
> I meant during it automatically as part of pip install (like in Hydra).

My current build code is similar to Hydra's: the parser code is indeed generated doing `pip install`.
The reason why this problem doesn't happen in Hydra is explained in https://github.com/omry/omegaconf/pull/321#discussion_r464077526

Reply:
roger